File: CSharp/1574.cs
using System;
using System.Linq;
using System.Collections.Generic;


class Program
{
    static void Main(string[] args)
    {
        int n = int.Parse(Console.ReadLine());
        int m;
        string movimento;
        List<string> movimentos = new List<string>();
        int ponto=0;

        for(int i =0;i<n;i++)
        {
            m = int.Parse(Console.ReadLine());
            for(int j =0;j<m;j++)
            {
                movimento =Console.ReadLine();
                if(movimento =="LEFT")
                {
                    ponto--;
                    movimentos.Add("LEFT");
                }
                else if(movimento =="RIGHT")
                {
                    ponto++;
                    movimentos.Add("RIGHT");
                }
                else
                {
                    movimento =movimentos[int.Parse(movimento.Split()[2])-1];
                    if(movimento =="LEFT")
                        {
                            ponto--;
                            movimentos.Add("LEFT");
                        }
                    else if(movimento =="RIGHT")
                        {
                            ponto++;
                            movimentos.Add("RIGHT");
                        }
                }
            }
            Console.WriteLine(ponto);
            movimentos.Clear();
            ponto=0;

        }
    }
}

File: Python/1574.py
n = int(input())
movimentos = []
ponto = 0

for _ in range(n):
    m = int(input())
    for _ in range(m):
        movimento = input()
        
        if movimento == "LEFT":
            ponto -= 1
            movimentos.append("LEFT")
        elif movimento == "RIGHT":
            ponto += 1
            movimentos.append("RIGHT")
        else:
            indice = int(movimento.split()[2]) - 1
            movimento = movimentos[indice]
            if movimento == "LEFT":
                ponto -= 1
                movimentos.append("LEFT")
            elif movimento == "RIGHT":
                ponto += 1
                movimentos.append("RIGHT")
                
    print(ponto)
    movimentos.clear()
    ponto = 0
